pkg/api-server/inspect_endpoints.go

```diff
@@ -102,10 +133,62 @@ func inspectDataplane(mpg core_xds.MatchedPoliciesGetter) restful.RouteFunction
 		}
 	}
 }
-func newDataplaneInspectResponse(matchedPolicies *core_xds.MatchedPolicies) []api_server_types.InspectEntry {
+
+func inspectPolicies(
+	resType core_model.ResourceType,
+	mpg core_xds.MatchedPoliciesGetter,
+	rm core_manager.ResourceManager,
+) restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		meshName := request.PathParameter("mesh")
+		policyName := request.PathParameter("name")
+
+		dataplanes := &core_mesh.DataplaneResourceList{}
+		if err := rm.List(context.Background(), dataplanes, store.ListByMesh(meshName)); err != nil {
+			rest_errors.HandleError(response, err, "Could not list Dataplanes")
+			return
+		}
+
+		result := []api_server_types.PolicyInspectEntry{}
+
+		for _, dp := range dataplanes.Items {
+			dpKey := core_model.MetaToResourceKey(dp.GetMeta())
+			matchedPolicies, err := mpg.Get(context.Background(), dpKey)
```

Comment:
I might be reading this wrong, but it seems like this may generate A LOT of DB queries.

`NewSimpleMatchedPolicyGetter` gets non cached ResourceManager.
Let's assume we have 1000 dataplanes in the system.
For every dataplane we run `mpg.Get` which not only fetches a single DP and Mesh but also all dataplanes in the mesh, so 1000 x (2 + 1000) = 1 002 000 queries. 

Even if we switch to cached ResourceManager I don't think it's ok.

Reply:
Updated PR to use MeshContextBuilder